packages/go_router/lib/src/route.dart

```diff
@@ -1024,6 +1042,22 @@ class StatefulShellBranch {
   /// The observers parameter is used by the [Navigator] built for this branch.
   final List<NavigatorObserver>? observers;
 
+  /// Whether this route branch should be loaded only when navigating to it for
+  /// the first time (the default behavior, i.e. 'false'), or if it should be
+  /// eagerly loaded (preloaded).
+  ///
+  /// If this property is true, the branch will be loaded immediately when the
+  /// associated [StatefulShellRoute] is visited for the first time. In that
+  /// case, the branch will be preloaded by navigating to the initial location
+  /// (see [initialLocation]).
+  ///
+  /// *Note:* The primary purpose of branch preloading is to enhance the user
+  /// experience when switching branches, which might for instance involve
+  /// preparing the UI for animated transitions etc. Care must be taken to
+  /// **keep the preloading to an absolute minimum** to avoid any unnecessary
```

Comment:
The position here is overly dramatic imo. I'd change 

```
 Care must be taken to **keep the preloading to an absolute minimum** to avoid any unnecessary resource use.
```

into 

```
 Preloading involves resource usage for elements that are not rendered, as such this should be used sparingly.
```

or something like that.

Reply:
Great feedback. This wording was actually due to some rather lengthy discussion in the original PR around the potential negative aspects of preloading, as I recall it. There are always dangers of premature/unnecessary optimisation when exposing these kinds of APIs, so some level of cautionary language is certainly justified. But maybe it can be dialed back a bit.